modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.http.internal.listener;
+
+import static org.mule.module.http.api.HttpHeaders.Names.CONTENT_LENGTH;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.mule.DefaultMuleMessage;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleMessage;
+import org.mule.module.http.internal.domain.response.HttpResponse;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class HttpResponseBuilderTestCase
+{
+
+    public static final String EXAMPLE_STRING = "exampleString";
+    private MuleContext muleContext;
+    private MuleMessage mockMuleMessage;
+    private MuleEvent mockEvent;
+
+    @Before
+    public void setUp()
+    {
+        muleContext = mock(MuleContext.class, RETURNS_DEEP_STUBS);
+        mockEvent = mock(MuleEvent.class);
+        mockMuleMessage = mock(MuleMessage.class);
+    }
+
+    @Test
+    public void testContentLenghtIsOverridden() throws Exception
+    {
+        HttpResponseBuilder httpResponseBuilder = new HttpResponseBuilder();
+        mockMuleMessage(new ByteArrayInputStream(EXAMPLE_STRING.getBytes(StandardCharsets.UTF_8)), 12);
+
+        HttpResponse httpResponse = httpResponseBuilder.build(new org.mule.module.http.internal.domain.response.HttpResponseBuilder(), mockEvent);
+        Assert.assertEquals("13", httpResponse.getHeaderValue(CONTENT_LENGTH));
```

Comment:
Had a hard time understanding the idea of the test. Can you use the length of the String and comment somewhere that you are setting it wrong on purpose?